app/controllers/storage_locations_controller.rb

```diff
@@ -18,9 +18,16 @@ def index
       @storage_locations = @storage_locations.kept
     end
 
+    inventory_item_names = []
+    InventoryItem.all.each{ |inv_item|
```

Comment:
This will fetch every single inventory item in the database. You probably want to do `@storage_locations.inventory_items`.

Also, in general we recommend using the `{ }` block syntax for single-line blocks, and `do..end` for multiline blocks.

Reply:
Thanks for the feedback! It's telling me that `@storage_locations.inventory_items ` isn't a method. Would something like this work? 
```
 @storage_locations.each do |storage_location| 
      storage_location.inventory_items.each do |inv_item|
        cur_item = Item.find(inv_item.item_id)
        inventory_item_names << cur_item.name if cur_item.active && !inventory_item_names.include?(cur_item.name)
      end
    end
```

Reply:
After going through your other comments, this is what I have now
```
inventory_item_names = []
    @storage_locations.each do |storage_location|
      inventory_item_names <<
        storage_location
        .get_active_items
        .joins(:item)
        .select('distinct items.name')
        .pluck(:name)
    end
    inventory_item_names.flatten!.uniq!.sort!
```